addons/html_builder/static/src/core/building_blocks/builder_range.js

```diff
@@ -18,17 +20,53 @@ export class BuilderRange extends Component {
         displayRangeValue: { type: Boolean, optional: true },
         computedOutput: { type: Function, optional: true },
         unit: { type: String, optional: true },
+        saveUnit: { type: String, optional: true },
+        applyWithUnit: { type: Boolean, optional: true },
+        withNumberInput: { type: Boolean, optional: true },
     };
     static defaultProps = {
         ...BuilderComponent.defaultProps,
         min: 0,
         max: 100,
         step: 1,
         displayRangeValue: false,
+        applyWithUnit: true,
+        withNumberInput: false,
     };
-    static components = { BuilderComponent };
+    static components = { BuilderComponent, BuilderNumberInput };
 
     setup() {
+        if (this.props.saveUnit && !this.props.unit) {
+            throw new Error("'unit' must be defined to use the 'saveUnit' props");
+        }
+
+        // this.syncedComponents keeps track of this component (BuilderRange),
+        // as well as the optional BuilderNumberInput. When a preview or commit
+        // happens on one of them, onBuilderComponentStateChange is called, and
+        // in case of a preview, the change is propagated to the other.
+        this.syncedComponents = new Set();
+        const registerSyncedBuilderComponent = (component) => {
+            if (!component) {
+                return () => {};
+            }
+            this.syncedComponents.add(component);
+            return () => {
+                this.syncedComponents.delete(component);
+            };
+        };
+        const onBuilderComponentStateChange = ({ component, reason, value } = {}) => {
+            if (!component || !this.syncedComponents.has(component)) {
+                return;
+            }
+            if (reason === "preview" && value !== undefined) {
+                this.propagatePreviewValue(value);
+            }
+        };
```

Comment:
I m not a big fan of this stuff. I need to think if I'm OK or if I have a better solution :)